modules/context/repo.go

```diff
@@ -426,10 +426,16 @@ func RepoAssignment() macaron.Handler {
 			return
 		}
 		ctx.Repo.GitRepo = gitRepo
+		defer func() {
+			if ctx.Repo.GitRepo == nil {
+				ctx.Repo.GitRepo.Close()
```

Comment:
`!= nil` ?

Reply:
oops.

Reply:
It would also help if I actually ran the defer function...